### PR TITLE
D3D12EnhancedBarriers: Fix markdown rendering issues and some typos

### PR DESCRIPTION
**Pull request by MarijnS95**
It seems that GitHub's view at https://github.com/microsoft/DirectX-Specs/blob/master/d3d/D3D12EnhancedBarriers.md is more lenient than the HTML rendered by `github-pages` hosted at https://microsoft.github.io/DirectX-Specs/d3d/D3D12EnhancedBarriers.html, resulting in e.g. bitwise OR's to appear as tables and ordered lists to be folded into a single line.

Also fix some grammar issues (while trying to keep sentence-rewrites to a minimum) and rename `D3D12_BARRIER_SYNC_RAYTRACING_ACCELERATION_STRUCTURE_{BUILD,COPY}` with BUILD/COPY inbetween `SYNC_` and `_RAYTRACING` instead.

### CODE REVIEW

File: d3d/D3D12EnhancedBarriers.md [hunk 1]
```diff
@@ -61,17 +61,17 @@ This document proposes an enhanced D3D12 Barrier API/DDI design that is capable
...
+  - [`ID3D12VideoDecodeCommandList3::Barrier()`](#id3d12videodecodecommandlist3barrier)
+  - [`ID3D12VideoProcessCommandList3::Barrier()`](#id3d12videoprocesscommandlist3barrier)
+  - [`ID3D12VideoEncodeCommandList3::Barrier()`](#id3d12videoencodecommandlist3barrier)
+  - [`ID3D12Device10::CreateCommittedResource3()`](#id3d12device10createcommittedresource3)
+  - [`ID3D12Device10::CreatePlacedResource2()`](#id3d12device10createplacedresource2)
+  - [`ID3D12Device10::CreateReservedResource2()`](#id3d12device10createreservedresource2)
+  - [`ID3D12DebugCommandQueue1::AssertResourceAccess()`](#id3d12debugcommandqueue1assertresourceaccess)
+  - [`ID3D12DebugCommandQueue1::AssertTextureLayout()`](#id3d12debugcommandqueue1asserttexturelayout)
+  - [`ID3D12DebugCommandList3::AssertResourceAccess()`](#id3d12debugcommandlist3assertresourceaccess)
+  - [`ID3D12DebugCommandList3::AssertTextureLayout()`](#id3d12debugcommandlist3asserttexturelayout)
```
Comment (vdwtanner) [inline d3d/D3D12EnhancedBarriers.md:74]: Hi, can you revert the change to the header names? We mirror these specs into an internal ADO repo as well, and ADO makes... _interesting_ choices about how it converts markdown headers to anchors. Making the header look nice ends up giving an anchor like this in ADO: `%60id3d12graphicscommandlist7%3A%3Abarrier()%60`. Other specs also link to some of these, so changing the anchor's can be hazardous even when it doesn't include special characters (ex: [Advanced Texture Ops](https://github.com/microsoft/DirectX-Specs/blob/master/d3d/HLSL_SM_6_7_Advanced_Texture_Ops.md?plain=1#L151-L153))

**Review by vdwtanner — CHANGES_REQUESTED** [review]
Thanks for the fixes! One change needed so that links work in ADO as well, then should be good to merge.

**Review by billkris-ms — APPROVED** [review]

File: d3d/D3D12EnhancedBarriers.md [hunk 2]
```diff
@@ -1279,7 +1281,7 @@ Some Access types require matching Sync.  For the following access bits, at leas
 
 ### `D3D12_BARRIER_LAYOUT`
 
-Describes any of the possible layouts used by D3D12 subresources. Layouts apply only to texture resources.  Buffer resources have only a linear layout, regardless of access type.
+Describes any of the possible layouts used by D3D12 subresources.  Layouts apply only to texture resources.  Buffer resources have only a linear layout, regardless of access type.
```
[the remark this answers is not in the document]
Comment (MarijnS95) [reply d3d/D3D12EnhancedBarriers.md:1286]: It seems tricky to revert these out, and double-spacing has clear preference: `\.  \w` is at 213 vs only 112 on `\. \w` (on this branch). Give or take the few `e.g. foo` occurences.

File: d3d/D3D12EnhancedBarriers.md [hunk 3]
```diff
@@ -764,9 +764,9 @@ Upload Heap resources are buffers and thus have no layout.  Upload Heap resource
 - `D3D12_BARRIER_ACCESS_SHADER_RESOURCE`
 - `D3D12_BARRIER_ACCESS_INDIRECT_ARGUMENT`
 - `D3D12_BARRIER_ACCESS_COPY_SOURCE`
-- `D3D12_BARRIER_ACCESS_RESOLVE_SOURCE` (*)
+- `D3D12_BARRIER_ACCESS_RESOLVE_SOURCE` [^1]
```
[the remark this answers is not in the document]
Comment (MarijnS95) [reply d3d/D3D12EnhancedBarriers.md:767]: I considered GitHub (and other markdown editors?) rendering them all at the end, but then nice (return) links are generated for them.

File: d3d/D3D12EnhancedBarriers.md (hunk 1 above)
Comment (MarijnS95) [reply d3d/D3D12EnhancedBarriers.md:74]: Good catch, I would have had to update those as well.  Reverted now and your suggestion made me discover some similarly broken links in `HLSL_SHaderModel6_7.md` and `VulkanOn12.md`.

Note that I couldn't find any link reference to https://github.com/microsoft/DirectX-Specs/blob/master/d3d/HLSL_SM_6_7_Advanced_Texture_Ops.md, is this file supposed to be reachable from anywhere (e.g. like `HLSL_SM_6_7_Wave_Ops_Include_Helper_Lanes.md`)?

Comment (vdwtanner) [reply d3d/D3D12EnhancedBarriers.md]: Oh nice, thanks for finding the others too!
Looks like Advanced Texture Ops should be reachable via homepage -> Shader Model 6.7 -> Advanced Texture Ops documentation link (relative address).

Comment (MarijnS95) [reply d3d/D3D12EnhancedBarriers.md]: I may have made a mess out of case-sensitive search then, the links are indeed there!

**Review by vdwtanner — APPROVED** [review]
Thanks!

**Pull request merged**